### DISCUSSION TITLE
Log which user triggered a dag

### DISCUSSION THREAD

**Discussion by tooptoop4** [apache/airflow#29592] [Ideas]
### Description

For audit purposes the username who triggered a dagrun should be logged (or stored in the metadb, shown in UI). Whether they logged into the UI and clicked trigger or used API to trigger a dagrun this should be captured.
ie userX triggered dagrunY with execDateZ
userA triggered dagrunB with execDateC

### Use case/motivation

Want to know who interacted with adhoc dagruns

### Related issues

no

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by doowhtron** [comment]
In the menu Browse -> logs, you may find what you need

**Comment by potiuk** [comment]
Related to #18217 - maybe you want to cooperate with @alex-astronomer and also add this one @tooptoop4 ?

**Comment by alex-astronomer** [comment]
Is this not already in place?  I know that we have audit logs for task clears, mark success, failure etc.  Off the cuff I'd be surprised if this didn't already exist, however I could be wrong.

**Comment by karunpoudel** [comment]
It would be helpful to add the triggered_by_user in the DagRun model itself, not just in Audit Log.
Use case:
From within a task, while the task is running, I have access to dag_run context variable from which I can know the external_trigger and run_type. If run_type = manual, then it's natural to ask which user triggered it.
triggered_by_user can be None if run_type is not manual

**Reply by uniqalex** [reply]
I am searching now solution for getting user which triggered my dag. And i agree with that it will be nice to have this info in dagRun context. 
And about audit Log - i have a case when some users triggered dag manually via web form and after that in audit log no record with event == ‘trigger’…

**Comment by karunpoudel** [comment]
Currently, Audit Log contains event type "trigger" but is it possible to tie a particular trigger event to a dag_run?

Is it possible to tie current running task's dag_run to its trigger event log so that I can get the user who triggered it? Lets say, the dag allows multiple active dag runs. A dag can be triggered at the same time by two different user with different config so it is important to tie those two correctly.

I was comparing the Audit Log's datetime and dag_run's logical_date and found those two to match  in 1 test. Can I assume they will always match to the same precision?
